File: my_py_pkg/my_py_pkg/smartPhone.py
#!/usr/bin/env python3

import rclpy
from rclpy.node import Node 
from example_interfaces.msg import String
class SubPhone(Node):
    def __init__(self):
        super().__init__("smartPhone")
        self.subscriber_ = self.create_subscription(String, "robot_news", self.subscriberCallback,10)
        self.get_logger().info("Phone Started")
    def subscriberCallback(self, msg):
        self.get_logger().info(msg.data)
    
def main(arg = None):
    rclpy.init(args = arg) 
    node = SubPhone()
    rclpy.spin(node)
    rclpy.shutdown()
if __name__ == "__main__":
    main()

File: my_cpp_pkg/src/smartphone.cpp
#include "example_interfaces/msg/string.hpp"
#include "rclcpp/rclcpp.hpp"

class SmartPhoneNode : public rclcpp::Node {

public:
  SmartPhoneNode() : Node("smartphone") {
    subscriber_ = this->create_subscription<example_interfaces::msg::String>(
        "robot_news", 10,
        std::bind(&SmartPhoneNode::callbackRobotNews, this,
                  std::placeholders::_1));
    RCLCPP_INFO(this->get_logger(), "Smartphone has bee started");
  }

private:
  void callbackRobotNews(const example_interfaces::msg::String::SharedPtr msg) {
    RCLCPP_INFO(this->get_logger(), "%s", msg->data.c_str());
  }

  rclcpp::Subscription<example_interfaces::msg::String>::SharedPtr subscriber_;
};

int main(int argc, char **argv) {
  rclcpp::init(argc, argv);
  auto node = std::make_shared<SmartPhoneNode>();
  rclcpp::spin(node);
  rclcpp::shutdown();
  return 0;
}